expressServer/index.js

```diff
@@ -12,55 +14,66 @@ const port = process.env.PORT || 5000;
 app.use(cors());
 app.use(express.json());
 
-const userSchema = new mongoose.Schema({
-  username: { type: String, required: true },
-  password: { type: String, required: true },
-});
-
-const User = mongoose.model('User', userSchema);
+const generateJWT = (user) => {
+  const payload = {
+    userId: user._id,
+    username: user.username,
+  };
+  const token = jwt.sign(payload, process.env.JWT_SECRET, { expiresIn: '24h' });
+  return token;
+};
 
 app.post('/api/login', async (req, res) => {
   const { username, password } = req.body;
 
   try {
-
     await connectDB();
 
-    const user = await User.findOne({ username, password });
-
+    const user = await User.findOne({ username });
     if (!user) {
       return res.status(401).json({ message: 'Incorrect username or password' });
     }
 
-    res.status(200).json({ message: 'Login successful', username });
+    const salt = user.salt;
+    const introducedHash = hashPassword(password, salt);
+
+    if (introducedHash !== user.hash) {
+      return res.status(401).json({ message: 'Incorrect username or password' });
+    }
+
+    const token = generateJWT(user);
+
+    res.status(200).json({ message: 'Login successful', token });
```

Comment:
You are generating a 'Bearer token', I see you directly return the token but this token does not include the type. The correct format is: 

```
{ message: 'Login successful', token: `Bearer ${token}` }